javascript/002-objetos-js-em-detalhe.md

```diff
@@ -3,91 +3,91 @@
 
-Aqui um simples objeto:
+Considere este simples objeto:
 
-```javascript
-var meuPrimeiroObjeto = { 
-	primeiroNome: "Richard",
-	autorFavorito: "Conrad" 
-}; 
+```js
+var meuPrimeiroObjeto = {primeiroNome: "Richard", autorFavorito: "Conrad"};
 ```
 
-Reiterando: Pense que um objeto é uma lista que contém itens e que cada item (uma propriedade) na lista é armazenada por um par nome-valor. Os nomes das propriedades no exemplo acima são `primeiroNome` e `autorFavorito`, e os valores para cada um "Richard" e "Conrad".
+Reiterando, pense num objeto como uma lista que contém ítens; e que cada item (uma propriedade ou um método) na lista é armazenado por um par nome-valor. Os nomes das propriedades no exemplo acima são `primeiroNome` e `autorFavorito`; e os valores são "Richard" e "Conrad".
 
-O nome das propriedades podem ser uma string ou um número, mas se for um número, ele deve ser acessado com o uso de colchetes. Veja no exemplo abaixo sobre esta notação, onde temos outro exemplo de objetos com números sendo o nome da propriedade.
+Os nomes das propriedades podem ser uma string ou um número; mas se o nome da propriedade for um número, ele tem de ser acessado com o uso de colchetes. Mais sobre a notação colchetes mais tarde. Aqui está um outro exemplo de objetos com números sendo o nome da propriedade:
 
 ```javascript
-var grupoDeIdade = { 30: "criança", 100: "muito velho" };
-
-console.log(grupoDeIdade.30) // Isso vai lançar um erro
+var grupoDeIdade = { 30: "Crianças", 100: "Muito Velho" };
+console.log(grupoDeIdade.30) // Isto vai lançar um erro
 
-// Assim é como você vai acessar o valor da propriedade 30, para que retorne o valor "criança"
-console.log(grupoDeIdade["30"]); //criança 
+// Assim é como você vai acessar o valor da propriedade 30, para obter o valor "Crianças"
+console.log(grupoDeIdade["30"]); // Crianças
 
 // É melhor evitar usar números como nomes de propriedades.
 ```
 
-Como um desenvolvedor JavaScript você frequentemente irá usar os tipos de dados Object, na maioria das vezes para armazenar dados e para criar seus próprios métodos e funções.
+Como um desenvolvedor JavaScript você muito freqüentemente irá usar os tipos de dados Object; na maioria das vezes para armazenar dados e para criar os seus próprios métodos e funções.
 
 
-## **Tipos de Dados por Referência e Tipos Primitivos de Dados**
-Uma das principais diferenças entre os tipos de dados por referência e os tipos primitivos de dados é que os dados por referência não são armazenados diretamente na variável como um valor primitivo é. Por exemplo:
+## **Tipo de Dados Referência e Tipos de Dados Primitivos**
+Uma das principais diferenças entre o tipo de dados referência e os tipos de dados primitivos é que o valor do tipo referência é armazenado como uma referência (ponteiro de memória). Ele não é portanto diretamente armazenado na variável como um valor, como os dos tipos primitivos são. Por exemplo:
```

Comment:
> One of the main differences between reference data type and primitive data types 
is reference data type’s value is stored as a reference, it is not stored 
directly on the variable, as a value, as the primitive data types are. 
For example:

A única coisa que acrescentei foi _(ponteiro de memória)_ para melhor elucidar o que vem a ser referência.

Foi também para quebrar um pouco a repetição da palavra _referência_ na mesma oração:

* _"é que o valor do tipo referência é armazenado como uma referência"_
* _"is reference data type’s value is stored as a reference,"_